tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/grpc/tool/StubGeneratorTestCase.java

```diff
@@ -333,6 +334,27 @@ public void testOneofFieldRecordGeneration() throws IllegalAccessException, Clas
                 "Expected imports not found in compile results.");
     }
 
+    @Test(description = "Test case checks creation of only the service file, in the service mode, with single service")
+    public void testServiceFileGenWithoutStub() throws IllegalAccessException, ClassNotFoundException,
+            InstantiationException {
+        Class<?> grpcCmd = Class.forName("org.ballerinalang.protobuf.cmd.GrpcCmd");
+        GrpcCmd grpcCommand = (GrpcCmd) grpcCmd.newInstance();
+        Path tempDirPath = outputDirPath.resolve("service");
+        Path protoPath = Paths.get("helloWorld.proto");
+        Path protoRoot = resourceDir.resolve(protoPath);
+        grpcCommand.setBalOutPath(tempDirPath.toAbsolutePath().toString());
+        grpcCommand.setProtoPath(protoRoot.toAbsolutePath().toString());
+        grpcCommand.setMode("service");
+        grpcCommand.execute();
+        Path sampleServiceFile = Paths.get(TMP_DIRECTORY_PATH, "grpc", "service", "helloWorld_sample_service.bal");
+
+        // This file should not be created when --mode service enabled with one service
+        Path sampleStubFile = Paths.get(TMP_DIRECTORY_PATH, "grpc", "service", "helloWorld_pb.bal");
+
+        assertTrue(Files.exists(sampleServiceFile));
+        assertFalse(Files.exists(sampleStubFile));
```

Comment:
Shall we compile the generated file and check whether there is any compilation error. and check whether message types and descriptor also there.
```java
CompileResult compileResult = BCompileUtil.compile(sampleServiceFile.toString());
assertEquals(compileResult.getDiagnostics().length, 0);
assertEquals(((BLangPackage) compileResult.getAST()).constants.size(), 1,
                "Expected constants not found in compile results.");
assertEquals(((BLangPackage) compileResult.getAST()).globalVars.size(), 1,
                "Expected global variables not found in compile results.");
```

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/21024/commits/355d18b3dc626ede2ae344231ac9889faab792f1